esignet-core/src/test/java/io/mosip/esignet/core/ValidatorTest.java

```diff
@@ -600,4 +601,35 @@ public void test_Signature_FormatValidator_withInvalidValue_thenFail() {
 		Assert.assertFalse(validator.isValid("ajjf", null));
 	}
 
+	//=========================== CodeChallengeValidator ==============================//
+
+	@Test
+	public void test_ValidCodeChallengeValidator_withValidDetails_thenPass(){
+		CodeChallengeValidator validator=new CodeChallengeValidator();
+		OAuthDetailRequestV2 request=new OAuthDetailRequestV2();
+		request.setCodeChallenge("codeChallenge");
+		request.setCodeChallengeMethod("codeChallengeMethod");
+		Assert.assertTrue(validator.isValid(request,null));
+		request.setCodeChallenge(null);
+		request.setCodeChallengeMethod(null);
+		Assert.assertTrue(validator.isValid(request,null));
+		request.setCodeChallenge("codeChallenge");
+		request.setCodeChallengeMethod(null);
+		Assert.assertFalse(validator.isValid(request,null));
```

Comment:
Why is the false assertion done in "_withValidDetails" test method?
We should add testcase for empty string as well kaif.

Reply:
done